File: php/features.php
<?php
declare(strict_types=1);

// HyruleCompendium SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class HyruleCompendiumFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new HyruleCompendiumBaseFeature();
            case "test":
                return new HyruleCompendiumTestFeature();
            default:
                return new HyruleCompendiumBaseFeature();
        }
    }
}

File: py/features.py
# HyruleCompendium SDK feature factory

from feature.base_feature import HyruleCompendiumBaseFeature
from feature.test_feature import HyruleCompendiumTestFeature


def _make_feature(name):
    features = {
        "base": lambda: HyruleCompendiumBaseFeature(),
        "test": lambda: HyruleCompendiumTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
